elasticgraph-warehouse/elasticgraph-warehouse.gemspec

```
@@ -0,0 +1,39 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require_relative "../elasticgraph-support/lib/elastic_graph/version"
+
+Gem::Specification.new do |spec|
+  spec.name = "elasticgraph-warehouse"
+  spec.version = ElasticGraph::VERSION
+  spec.authors = ["Myron Marston", "Ben VandenBos", "Block Engineering"]
+  spec.email = ["myron@squareup.com"]
+  spec.homepage = "https://block.github.io/elasticgraph/"
+  spec.license = "MIT"
+  spec.summary = "Extends ElasticGraph to support ingestion into a data warehouse."
+
+  spec.metadata = {
+    "bug_tracker_uri" => "https://github.com/block/elasticgraph/issues",
+    "changelog_uri" => "https://github.com/block/elasticgraph/releases/tag/v#{ElasticGraph::VERSION}",
+    "documentation_uri" => "https://block.github.io/elasticgraph/api-docs/v#{ElasticGraph::VERSION}/",
+    "homepage_uri" => "https://block.github.io/elasticgraph/",
+    "source_code_uri" => "https://github.com/block/elasticgraph/tree/v#{ElasticGraph::VERSION}/#{spec.name}",
+    "gem_category" => "extension"
+  }
+
+  spec.files = Dir.chdir(File.expand_path(__dir__)) do
+    `git ls-files -z`.split("\x0").reject do |f|
+      (f == __FILE__) || f.match(%r{\A(?:(?:test|spec|features|sig)/|\.(?:git|travis|circleci)|appveyor)})
+    end - [".rspec", "Gemfile", ".yardopts"]
+  end
+
+  spec.required_ruby_version = [">= 3.4", "< 3.5"]
+
+  spec.add_dependency "elasticgraph-schema_definition", ElasticGraph::VERSION
```

```suggestion
  spec.add_development_dependency "elasticgraph-schema_definition", ElasticGraph::VERSION
```

One bit of feedback I was planning to provide on #863 was to move `indexer_extension.rb` and `warehouse_dumper.rb` into this gem from the `elasticgraph-warehouse_lambda` gem. The `*-lambda` gems are ideally very thin wrappers around logic that lives in non-lambda gems so that all parts of ElasticGraph are usable outside of a lambda context.

With that in mind, the non-lambda parts of #863 would live in this gem, and this gem would get deployed to production. We don't want `elasticgraph-schema_definition` pulled in and also deployed to prod (which would happen if it's a normal dependency here). As a `development_dependency`, it's available for use in the test suite without being pulled in when users include this gem in their production bundle. Besides, we can trust that `elasticgraph-schema_definition` will already be part of the local bundle for ElasticGraph users as it gets pulled in already.